Added support for commonjs module

seidelmartin opened the pull request:
Added babel transformation that produces commonjs module so it can be used both as esmodule and commonjs module.

gullerya commented on ci/tools/build-utils.js:
```diff
@@ -1,66 +1,124 @@
+const SRC_DIR = 'src';
+const DIST_DIR = 'dist';
+
+const buildCDN = process.argv.some(a => a === '--cdn')
```
```suggestion
const buildCDN = process.argv.some(a => a === '--cdn');
```

gullerya commented on ci/tools/build-utils.js:
```diff
@@ -1,66 +1,124 @@
 
-process.stdout.write(`\x1B[32mStarting the build...\x1B[0m${os.EOL}`);
-process.stdout.write(os.EOL);
+stdout.writeGreen('Starting the build...')
```
```suggestion
stdout.writeGreen('Starting the build...');
```

gullerya commented on ci/tools/build-utils.js:
```diff
@@ -1,66 +1,124 @@
+stdout.writeNewline();
 
-await ensureCleanDir(DIST);
+await cleanDistDir()
```
```suggestion
await cleanDistDir();
```

gullerya commented on ci/tools/build-utils.js:
```diff
@@ -1,66 +1,124 @@
+		cleanDist: false,
+		srcDir: SRC_DIR,
+		distDir: path.join(DIST_DIR),
+		ext: 'mjs',
```
Probably this can stay `js` as well? The package has the type as module, so should be treated as ESM module by default.

seidelmartin replied:
Yes, I haven't been sure about this so I left it as `mjs` but will change it to `js`.

gullerya commented on ci/tools/build-utils.js:
```diff
@@ -1,66 +1,124 @@
+		distDir: path.join(DIST_DIR),
+		ext: 'mjs',
+		format: 'esm',
+	});
```
add pattern property just for symmetry with the CJS builder?

seidelmartin replied:
This way it also copies the `d.ts` file but I can make that pattern so it takes it as well.

gullerya commented on ci/tools/build-utils.js:
```diff
@@ -1,66 +1,124 @@
-}
\ No newline at end of file
+
+	const sriMap = await calcIntegrity(CDN_DIR)
```
```suggestion
	const sriMap = await calcIntegrity(CDN_DIR);
```

gullerya commented:
@seidelmartin Looks good, the build script looks in much better order now for sure!
Left few minor styling remarks and the only one which might be conceptual is about - why not leaving the ESM as `js` and not `mjs`?

seidelmartin commented:
@gullerya I've addressed those comments. Can you please check? Is there anything else I should do? Changelog record, bump the version or something? :)

gullerya commented:
@seidelmartin perfect, approved, you can merge (not sure if you have that allowed, if not, let me know I'll merge).

Bump is done automatically during the release, no need to do that.

Changelog - you can add a reference to the unreleased area, just see the examples in prev releases as an example, if not - I'll do it, however you wish.

I'll also update the readme part.

It might take me a couple of days to release it, I'll update in the ticket when done.

seidelmartin commented:
@gullerya I've added the changelog record but I cannot merge because I don't have write permissions.
